src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -415,17 +418,21 @@ def unique(self, level: Hashable | None = None) -> Index:
         See Also
         --------
         unique : Numpy array of unique values in that column.
-        Series.unique : Return unique values of Series object.
+        Series.unique : Return unique values of a Series object.
 
         Examples
         --------
         >>> idx = pd.Index([1, 1, 2, 3, 3])
         >>> idx.unique()
         Index([1, 2, 3], dtype='int64')
         """
-        # TODO: SNOW-1458132 implement unique
-        WarningMessage.index_to_pandas_warning("unique")
-        return Index(self.to_pandas().unique(level=level))
+        # TODO: SNOW-1514782 MultiIndex - support level for Index.unique.
+        if level is not None:
+            ErrorMessage.not_implemented(
+                "level is not yet supported for Index.unique. It is used with "
+                "MultiIndex objects, which will be supported in the future."
+            )
+        return Index(data=self._query_compiler.unique(is_index=True, level=0))
```

Comment:
```suggestion
        result = Index(data=self._query_compiler.reset_index(names=['index']).unique().set_index(['index']))
        result.name = self.name
        return result
```

I think you can implement `unique()` this way without changing the query compiler `unique()` method:

Reply:
I'm keeping the original logic I have here